core/src/main/scala/com/avast/clients/rabbitmq/StreamingRabbitMQConsumer.scala

```diff
@@ -0,0 +1,272 @@
+package com.avast.clients.rabbitmq
+
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicReference}
+
+import cats.effect.concurrent.{Deferred, Semaphore}
+import cats.effect.{Blocker, CancelToken, ConcurrentEffect, ContextShift, Effect, ExitCase, IO, Resource, Sync}
+import cats.syntax.all._
+import com.avast.bytes.Bytes
+import com.avast.clients.rabbitmq.DefaultRabbitMQConsumer.RepublishOriginalRoutingKeyHeaderName
+import com.avast.clients.rabbitmq.DefaultStreamingRabbitMQConsumer.DeliveryQueue
+import com.avast.clients.rabbitmq.api.{Delivery, DeliveryResult}
+import com.avast.metrics.scalaapi.Monitor
+import com.rabbitmq.client._
+import com.typesafe.scalalogging.StrictLogging
+import fs2.Stream
+import fs2.concurrent.Queue
+
+import scala.language.higherKinds
+import scala.util.control.NonFatal
+
+trait StreamingRabbitMQConsumer[F[_], A] {
```

Comment:
Shouldn't these traits be in a separate file?

Reply:
Good point.

Reply:
Unfortunately, it's not that easy :-D It can surely be split into separate file, but I just have realized why I didn't do that - it would make sense to have the trait in `api` module, but then the `api` module would need to depend on the `fs2` library, because there is an `fs2.Stream` as part of a return type in the `StreamingRabbitMQConsumer`. Do you think it's worth it to have it just separated but not moved to the `api` module?

Reply:
It is part of your API so I am afraid you need to add it to the `api` module. Otherwise it does not make sense.